### PR TITLE
fix(react): enable event names with forward slash

### LINKED ISSUE #473: bug(react-output-target): eventName with special characters results in invalid code generation
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/.github/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil-ds-output-targets/issues) that already report this problem, without success.

### Stencil Version

4.21

### Stencil Framework Output Target

React

### Stencil Framework Output Target Version

0.7.1

### Current Behavior

image

### Expected Behavior

valid typesccript

### Steps to Reproduce

1. Create a component with an event
  @Event({
    eventName: 'foo/bar',
  })
  
 2. build stencil with react-output-target
 
 

### Code Reproduction URL

github.com 

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by alexprudhomme**
fix #473 

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally for affected output targets
- [x] Tests (`npm test`) were run locally and passed
- [x] Prettier (`npm run prettier`) was run locally and passed

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue URL:

## What is the new behavior?

- enables event names with forward slash in react-output-target
-
-

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by tanner-reits — CHANGES_REQUESTED** [review]
@alexprudhomme I tested this out with your `/` event naming, but doesn't seem to work with that convention. As far as I know, React requires strict camel-casing for event handlers, so if you wanted to have event names with a `/` in them, you would need to update the logic [here](https://github.com/ionic-team/stencil-ds-output-targets/blob/main/packages/react-output-target/src/react-output-target/utils/string-utils.ts#L10) to also perform those replacements

**Comment by alexprudhomme** [comment]
> @alexprudhomme I tested this out with your `/` event naming, but doesn't seem to work with that convention. As far as I know, React requires strict camel-casing for event handlers, so if you wanted to have event names with a `/` in them, you would need to update the logic [here](https://github.com/ionic-team/stencil-ds-output-targets/blob/main/packages/react-output-target/src/react-output-target/utils/string-utils.ts#L10) to also perform those replacements

Hey, thanks for the useful tip. I changed the implementation and added a test. Let me know if you need anything more.

**Review by tanner-reits — CHANGES_REQUESTED** [review]
@alexprudhomme Functionality looks good! Only thing I would like to see changed is to leave the `kebabToCamelCase` function as it was (in case we have need for that utility later), and move the `/` replacement to a separate utility function that can be called in `eventListenerName`.

File: packages/react-output-target/src/react-output-target/utils/string-utils.ts [hunk 1]
```diff
@@ -7,4 +7,9 @@ export const kebabToPascalCase = (str: string) =>
 
 export const kebabToCamelCase = (str: string) => str.replace(/-([a-z])/g, (_, letter) => letter.toUpperCase());
 
-export const eventListenerName = (eventName: string) => kebabToCamelCase(`on-${eventName}`);
+const dashesToCamelCase = (str: string) => str.replace(/\/([a-z])/g, (_, letter) => letter.toUpperCase());
```
Comment (tanner-reits) [inline packages/react-output-target/src/react-output-target/utils/string-utils.ts:10]:
```suggestion
const slashesToCamelCase = (str: string) => str.replace(/\/([a-z])/g, (_, letter) => letter.toUpperCase());
```

File: packages/react-output-target/src/react-output-target/utils/string-utils.ts [hunk 2]
```diff
@@ -7,4 +7,9 @@ export const kebabToPascalCase = (str: string) =>
...
+
+export const eventListenerName = (eventName: string) => {
+  const dashedConverted = dashesToCamelCase(eventName);
+  return kebabToCamelCase(`on-${dashedConverted}`);
```
Comment (tanner-reits) [inline packages/react-output-target/src/react-output-target/utils/string-utils.ts:14]:
```suggestion
  const slashesConverted = slashesToCamelCase(eventName);
  return kebabToCamelCase(`on-${slashesConverted}`);
```

**Review by tanner-reits — CHANGES_REQUESTED** [review]
Just some naming fixes, then g2g

**Review by tanner-reits — APPROVED** [review]

**Comment by alexprudhomme** [comment]
@tanner-reits 
Hi, 
Will you be able to make a release soon with this fix ? Last one was on august 28th. 
Thanks !

**Comment by christian-bromann** [comment]
@alexprudhomme I am working on it as we speak.

**Pull request merged**